core/infrastructure/imagestream_reconciler.go

```diff
@@ -0,0 +1,167 @@
+	if err != nil {
+		return
+	}
+
+	// Process Delta
+	return i.processDelta(requestedResources, deployedResources)
+}
+
+func (i *imageStreamReconciler) createRequiredResources() (map[reflect.Type][]resource.KubernetesResource, error) {
+	resources := make(map[reflect.Type][]resource.KubernetesResource)
+	imageStreamHandler := NewImageStreamHandler(i.Context)
+	imageStream, err := imageStreamHandler.CreateImageStreamIfNotExists(i.key, i.tag, i.addFromReference, i.imageName, i.insecureImageRegistry)
+	if err != nil {
+		return nil, err
+	}
+	if imageStream != nil {
+		resources[reflect.TypeOf(imgv1.ImageStream{})] = []resource.KubernetesResource{imageStream}
+	}
+	if err := i.setOwner(resources); err != nil {
+		return resources, err
+	}
+	return resources, nil
+}
+
+func (i *imageStreamReconciler) getDeployedResources() (map[reflect.Type][]resource.KubernetesResource, error) {
+	resources := make(map[reflect.Type][]resource.KubernetesResource)
+	imageStreamHandler := NewImageStreamHandler(i.Context)
+	// fetch owned image stream
+	ownedImageStream, err := imageStreamHandler.FetchImageStreamForOwner(i.owner)
+	if err != nil {
+		return nil, err
+	}
+	if ownedImageStream != nil {
+		resources[reflect.TypeOf(imgv1.ImageStream{})] = append(resources[reflect.TypeOf(imgv1.ImageStream{})], ownedImageStream...)
+	}
+
+	// fetch the shared image
+	sharedImageStream, err := imageStreamHandler.FetchImageStream(i.key)
+	if err != nil {
+		return nil, err
+	}
+	if sharedImageStream != nil {
+		resources[reflect.TypeOf(imgv1.ImageStream{})] = append(resources[reflect.TypeOf(imgv1.ImageStream{})], sharedImageStream)
+	}
+	return resources, nil
+}
+
+func (i *imageStreamReconciler) processDelta(requestedResources map[reflect.Type][]resource.KubernetesResource, deployedResources map[reflect.Type][]resource.KubernetesResource) (reconcileInterval time.Duration, err error) {
+	comparator := i.getComparator()
+	deltas := comparator.Compare(deployedResources, requestedResources)
+	isDeltaProcessed := false
+	for resourceType, delta := range deltas {
+		if !delta.HasChanges() {
+			i.Log.Info("No delta found", "resourceType", resourceType)
+			continue
+		}
+		i.Log.Info("Will", "create", len(delta.Added), "update", len(delta.Updated), "delete", len(delta.Removed), "resourceType", resourceType)
+
+		if _, err = kubernetes.ResourceC(i.Client).CreateResources(delta.Added); err != nil {
+			return
+		}
+
+		if _, err = kubernetes.ResourceC(i.Client).UpdateResources(deployedResources[resourceType], delta.Updated); err != nil {
+			return
+		}
+
+		if _, err = kubernetes.ResourceC(i.Client).DeleteResources(delta.Removed); err != nil {
+			return
+		}
+
+		isDeltaProcessed = true
+	}
+
+	if isDeltaProcessed {
+		return time.Second * 10, nil
+	}
+	return
+}
+
+// setOwner sets this service instance as the owner of each resource.
+func (i *imageStreamReconciler) setOwner(resources map[reflect.Type][]resource.KubernetesResource) error {
```

Comment:
I would move that method to `infrastructure/image_stream.go` file and add a `AddOwnerToImageStream` or similar name
so that we have all imagestream owner manipulation in the same place
wdyt ?

Reply:
This method very generic and doesn't contain any ImageStream specific code. It just loop over all kubernetes resources and call `framework.AddOwnerReference` method.
`infrastructure/image_stream.go` is more related to image stream related operations.